From: Jiri Olsa <jolsa@kernel.org>
Date: Wed Jan 20 12:56:34 2016 +0100
Subject: perf stat: Do not clean event's private stats
Git-commit: 3f416f22d1e21709a631189ba169f76fd267b374
Patch-mainline: v4.5-rc2
References: bsc#967690
Signed-off-by: Tony Jones <tonyj@suse.de>

    perf stat: Do not clean event's private stats
    
    Mel reported stddev reporting was broken due to following commit:
    
    	106a94a0f8c2 ("perf stat: Introduce read_counters function")
    
    This commit merged interval and overall counters reading into single
    read_counters function.
    
    The old interval code cleaned the stddev data for some reason (it's
    never displayed in interval mode) and the mentioned commit kept on
    cleaning the stddev data in merged function, which resulted in the
    stddev not being displayed.
    
    Removing the wrong stddev data cleanup init_stats call.
    
    Reported-and-Tested-by: Mel Gorman <mgorman@techsingularity.net>
    Signed-off-by: Jiri Olsa <jolsa@kernel.org>
    Cc: David Ahern <dsahern@gmail.com>
    Cc: Namhyung Kim <namhyung@kernel.org>
    Cc: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Cc: stable@vger.kernel.org # v4.2+
    Fixes: 106a94a0f8c2 ("perf stat: Introduce read_counters function")
    Link: http://lkml.kernel.org/r/1453290995-18485-4-git-send-email-jolsa@kernel.org
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>

diff --git a/tools/perf/util/stat.c b/tools/perf/util/stat.c
index 2f901d1..2b58edc 100644
--- a/tools/perf/util/stat.c
+++ b/tools/perf/util/stat.c
@@ -310,7 +310,6 @@ int perf_stat_process_counter(struct perf_stat_config *config,
 	int i, ret;
 
 	aggr->val = aggr->ena = aggr->run = 0;
-	init_stats(ps->res_stats);
 
 	if (counter->per_pkg)
 		zero_per_pkg(counter);
